spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/logging/AdaptableLogger.java

```diff
@@ -0,0 +1,32 @@
+package org.springframework.cloud.gateway.logging;
+
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * An abstraction over logging. For example, to maintain proper trace context when using
+ * manual sleuth instrumentation the log call needs to be wrapped inside a special WebFlux
+ * operator.
+ */
+public interface AdaptableLogger {
```

Comment:
I guess we need proper javadocs + file header with copyrights here (I guess this is why you've disabled checkstyle plugin)